### PR TITLE
[ID1000295] define struct

### PR DESCRIPTION
**Pull request by hukunge**
--story=ID1000295 --user=hukunge
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000295

Signed-off-by: hukunge <hukunge@163.com>

### CODE REVIEW

**Comment by hukunge** [comment]
--story=ID1000295 --user=hukunge
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000295

File: models/models.go [hunk 1]
```diff
@@ -7,13 +7,153 @@ SPDX-License-Identifier: Apache-2.0
...
+	KycStatus      string `gorm:"type:varchar(16)"`              // kyc status
+	Bank           string `gorm:"type:varchar(64)"`              // bank name
+	BankCardNum    string `gorm:"type:varchar(64)"`              // bank card num
+	TaxID          string `gorm:"type:varchar(128)"`             // tax id
+	ShippingAddrID string `gorm:"type:varchar(256)"`             // shipping address id
+	Did            string `gorm:"type:varchar(64)"`              // did
+	Desc           string `gorm:"size:1024"`                     // description
+	CreatedAt      time.Time
+	UpdatedAt      time.Time
+	DeletedAt      time.Time
```
Comment (zhangbing16) [inline models/models.go]: DeletedAt *time.Time `sql:"indel"`

Comment (zhangbing16) [inline models/models.go]: DeletedAt *time.Time `sql:"indel"`
所有的表中都要修改

File: models/models.go [hunk 2]
```diff
@@ -7,13 +7,153 @@ SPDX-License-Identifier: Apache-2.0
...
+	Type           string `gorm:"type:varchar(16)"`              // user type
+	Phone          string `gorm:"type:varchar(32)"`              // phone num
+	Email          string `gorm:"type:varchar(128);unique"`      // email
+	KycStatus      string `gorm:"type:varchar(16)"`              // kyc status
+	Bank           string `gorm:"type:varchar(64)"`              // bank name
+	BankCardNum    string `gorm:"type:varchar(64)"`              // bank card num
+	TaxID          string `gorm:"type:varchar(128)"`             // tax id
+	ShippingAddrID string `gorm:"type:varchar(256)"`             // shipping address id
+	Did            string `gorm:"type:varchar(64)"`              // did
+	Desc           string `gorm:"size:1024"`                     // description
```
Comment (zhangbing16) [inline models/models.go]: `gorm:"text"`

File: models/models.go [hunk 3]
```diff
@@ -7,13 +7,153 @@ SPDX-License-Identifier: Apache-2.0
...
-// Account account model
+// Account defines the common information of user
 type Account struct {
-	ID        string
+	ID             string `gorm:"type:varchar(256);primary_key"` // user id
+	Password       string `gorm:"type:varchar(256)"`             // password
+	NickName       string `gorm:"type:varchar(64)"`              // nick name
+	Type           string `gorm:"type:varchar(16)"`              // user type
+	Phone          string `gorm:"type:varchar(32)"`              // phone num
+	Email          string `gorm:"type:varchar(128);unique"`      // email
```
Comment (zhangbing16) [inline models/models.go]: 先不考虑unique

File: models/models.go [hunk 4]
```diff
@@ -7,13 +7,153 @@ SPDX-License-Identifier: Apache-2.0
...
 }
+
+// DonationStat defines the donation statistics of organization
+type DonationStat struct {
+	ID               string          `gorm:"type:varchar(256);primary_key"` // donation statistics id
+	UID              string          `gorm:"type:varchar(256);not null"`    // user id
+	ReceivingSupply  int64           // receiving supply
+	DistributeSupply int64           // distribute supply
+	ReceivingFunds   decimal.Decimal `gorm:"type:decimal(30,4)"` // receiving funds
+	DistributeFunds  decimal.Decimal `gorm:"type:decimal(30,4)"` // distribute funds
```
Comment (zhangbing16) [inline models/models.go]: ReceivedSupplies?
DistributedSupplies?
ReceivedFunds?
DistributedFunds?

File: models/models.go [hunk 5]
```diff
@@ -7,13 +7,153 @@ SPDX-License-Identifier: Apache-2.0
...
+	Status      string `gorm:"type:varchar(32)"`              // the status of certification
+	Expired     int64  // expired time
+	Remark      string `gorm:"size:1024"` // remark
+	CreatedAt   time.Time
+	UpdatedAt   time.Time
+	DeletedAt   time.Time
+}
+
+// Photo defines user's photo
+type Photo struct {
```
Comment (zhangbing16) [inline models/models.go]: Image?

File: models/models.go [hunk 6]
```diff
@@ -7,13 +7,153 @@ SPDX-License-Identifier: Apache-2.0
...
+	ID        string `gorm:"type:varchar(256);primary_key"` // phone id
```
Comment (zhangbing16) [inline models/models.go]: comments is “image id” ？

File: models/models.go [hunk 7]
```diff
@@ -7,13 +7,153 @@ SPDX-License-Identifier: Apache-2.0
...
+	RelatedID string `gorm:"type:varchar(256);not null"`    // the related id
+	Type      string `gorm:"type:varchar(64)"`              // user type
+	URL       string `gorm:"type:varchar(256)"`             // phone url
```
Comment (zhangbing16) [inline models/models.go]: comments is ”image url“？

File: models/models.go [hunk 8]
```diff
@@ -7,13 +7,153 @@ SPDX-License-Identifier: Apache-2.0
...
+	UID         string          `gorm:"type:varchar(256);not null"`    // user id
+	AidUID      string          `gorm:"type:varchar(256)"`             // aid user id
+	TargetUID   string          `gorm:"type:varchar(256)"`             // user id of charity
+	PubType     string          `gorm:"type:varchar(16)"`              // the type of publicity
+	PayType     string          `gorm:"type:varchar(16)"`              // pay type
+	Amount      decimal.Decimal `gorm:"type:decimal(30,4)"`            // the amount of publicity funds
+	TxID        string          `gorm:"type:varchar(256)"`             // block chain tx id
+	Remark      string          `gorm:"size:1024"`                     // remark
+	BlockType   string          `gorm:"type:varchar(32)"`              // block type
+	BlockHeight int             // block height
```
Comment (zhangbing16) [inline models/models.go]: int64？

File: models/models.go [hunk 9]
```diff
@@ -7,13 +7,153 @@ SPDX-License-Identifier: Apache-2.0
...
+	AidUID      string `gorm:"type:varchar(256)"`             // aid user id
+	TargetUID   string `gorm:"type:varchar(256)"`             // user id of charity
+	PubType     string `gorm:"type:varchar(16)"`              // the type of publicity
+	Name        string `gorm:"type:varchar(512)"`             // name
+	Number      int64  // number
+	Unit        string `gorm:"type:varchar(32)"`  // unit
+	TxID        string `gorm:"type:varchar(256)"` // block chain tx id
+	Remark      string `gorm:"size:1024"`         // remark
+	BlockType   string `gorm:"type:varchar(32)"`  // block type
+	BlockHeight int    // block height
```
Comment (zhangbing16) [inline models/models.go]: int64？

**Pull request closed without merging**